sound/soc/sof/pcm.c

```diff
@@ -413,8 +413,8 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 	if (ipc_first)
 		snd_sof_pcm_platform_trigger(sdev, substream, cmd);
 
-	/* free PCM if reset_hw_params is set and the STOP IPC is successful */
-	if (!ret && reset_hw_params)
+	/* free PCM if pcm_free is set and the STOP IPC is successful */
+	if (!ret && pcm_free)
```

Comment:
It took me a while to understand your change, and I guess it's mainly fixing the 'fallthrough'

I would update the commit message to make it clearer.

Reply:
@ranj063 I need more convincing. The fallthrough was not accident and reset_hw_params was set on purpose. At least back then, the only way to reset the DSP state was to send PCM_FREE. 

Unless I miss something, this change just brings the old bug back.